### PR TITLE
[MINOR] chore(ci): Add a shade check step after build spark shade jar

### PR DESCRIPTION
**Pull request by maobaolong**
### What changes were proposed in this pull request?

Add a check after build spark shaded plugin, to check whether there are new coming unshaded content.

### Why are the changes needed?

Without this, we cannot prevent to merge some PR which will break the shade client.

### Does this PR introduce _any_ user-facing change?

No.

### How was this patch tested?

CI Tests.

### CODE REVIEW

**Comment by maobaolong** [comment]
@EnricoMi Would you like to take a look at this PR? Thank you!

File: dev/scripts/checkshade.sh [hunk 1]
```diff
@@ -0,0 +1,39 @@
...
+# limitations under the License.
+#
+
+set -o pipefail
+set -o nounset   # exit the script if you try to use an uninitialised variable
+
+SHADED_JAR_PATH=${1}
+echo "checkShaded:"${SHADED_JAR_PATH}
+if [ ! -f "${SHADED_JAR_PATH}" ]; then
+    echo "check failed, file is not exist."
```
Comment (EnricoMi) [inline dev/scripts/checkshade.sh:26]:
```suggestion
    echo "check failed, file does not exist."
```

File: dev/scripts/checkshade.sh [hunk 2]
```diff
@@ -0,0 +1,39 @@
...
+    exit -1
```
Comment (EnricoMi) [inline dev/scripts/checkshade.sh:27]: non-zero exit code is sufficient, negative codes are not common / needed
```suggestion
    exit 1
```

File: dev/scripts/checkshade.sh [hunk 3]
```diff
@@ -0,0 +1,39 @@
...
+fi
+
+UNSHADED_COUNT=`unzip -l $1 | awk 'NR>3 {print \$4}' | grep -vE 'uniffle|shuffle|META-INF|git.properties|/$|\.html$|\.css$|^javax|^$' | wc -l`
+echo "unshaded count = $UNSHADED_COUNT"
+echo "content is :"
+echo `unzip -l $1 | awk 'NR>3 {print \$4}' | grep -vE 'uniffle|shuffle|META-INF|git.properties|/$|\.html$|\.css$|^javax|^$'`
```
Comment (EnricoMi) [inline dev/scripts/checkshade.sh:33]:
```suggestion
UNSHADED=$(unzip -l "$1" | awk 'NR>3 {print \$4}' | grep -vE 'uniffle|shuffle|META-INF|git.properties|/$|\.html$|\.css$|^javax|^$')
UNSHADED_COUNT=$(wc -l <<< "$UNSHADED")
echo "unshaded count: $UNSHADED_COUNT"
echo "unshaded content:"
echo "$UNSHADED"
```

File: dev/scripts/checkshade.sh [hunk 4]
```diff
@@ -0,0 +1,39 @@
...
+if [ $UNSHADED_COUNT -le 1 ]; then
+    echo "check success."
+else
+    echo "check failed."
+    exit -2
```
Comment (EnricoMi) [inline dev/scripts/checkshade.sh:38]:
```suggestion
    exit 2
```

File: dev/scripts/checkshade.sh [hunk 5]
```diff
@@ -0,0 +1,39 @@
...
+if [ ! -f "${SHADED_JAR_PATH}" ]; then
+    echo "check failed, file is not exist."
+    exit -1
+fi
+
+UNSHADED_COUNT=`unzip -l $1 | awk 'NR>3 {print \$4}' | grep -vE 'uniffle|shuffle|META-INF|git.properties|/$|\.html$|\.css$|^javax|^$' | wc -l`
+echo "unshaded count = $UNSHADED_COUNT"
+echo "content is :"
+echo `unzip -l $1 | awk 'NR>3 {print \$4}' | grep -vE 'uniffle|shuffle|META-INF|git.properties|/$|\.html$|\.css$|^javax|^$'`
+if [ $UNSHADED_COUNT -le 1 ]; then
```
Comment (EnricoMi) [inline dev/scripts/checkshade.sh:34]: Why is an unshaded count of `1` successful?

Comment (maobaolong) [reply dev/scripts/checkshade.sh]: Updated

**Comment by maobaolong** [comment]
@EnricoMi Thanks for your suggestion, it make this script looks better than before, PTAL.

**Comment by EnricoMi** [comment]
Why have you removed the `-o pipefail`?

File: client-spark/spark2-shaded/pom.xml [hunk 6]
```diff
@@ -312,6 +312,9 @@
                 <echo message="repackaging netty jar"></echo>
                 <jar destfile="${project.build.directory}/${project.artifactId}-${project.version}.jar"
                      basedir="${project.build.directory}/unpacked"/>
+                <exec executable="sh" failonerror="true" dir="${project.build.directory}/">
```
Comment (EnricoMi) [inline client-spark/spark2-shaded/pom.xml:315]: Use `bash` shell and the `-o pipefail` shoudl work:
```suggestion
                <exec executable="bash" failonerror="true" dir="${project.build.directory}/">
```

File: client-spark/spark3-shaded/pom.xml [hunk 7]
```diff
@@ -309,6 +309,9 @@
                 <echo message="repackaging netty jar"></echo>
                 <jar destfile="${project.build.directory}/${project.artifactId}-${project.version}.jar"
                      basedir="${project.build.directory}/unpacked"/>
+                <exec executable="sh" failonerror="true" dir="${project.build.directory}/">
```
Comment (EnricoMi) [inline client-spark/spark3-shaded/pom.xml:312]:
```suggestion
                <exec executable="bash" failonerror="true" dir="${project.build.directory}/">
```

**Review by EnricoMi — CHANGES_REQUESTED** [review]
Please restore the `-o pipefail` and use `bash` shell.

**Review by EnricoMi — APPROVED** [review]
LGTM!

**Comment by maobaolong** [comment]
@EnricoMi Thanks for your review!

**Pull request merged**